lustrefs-exporter/src/brw_stats.rs

```diff
@@ -1,495 +1,522 @@
+                    .build(),
+                recovery_status_evicted_clients: meter
+                    .u64_gauge("recovery_status_evicted_clients")
+                    .with_description("Gives the count of clients evicted from a target.")
+                    .build(),
+                ost_stats: meter
+                    .u64_gauge("lustre_oss_ost_stats")
+                    .with_description("OSS ost stats")
+                    .build(),
+                ost_io_stats: meter
+                    .u64_gauge("lustre_oss_ost_io_stats")
+                    .with_description("OSS ost_io stats")
+                    .build(),
+                ost_create_stats: meter
+                    .u64_gauge("lustre_oss_ost_create_stats")
+                    .with_description("OSS ost_create stats")
+                    .build(),
+                changelog_current_index: meter
+                    .u64_gauge("lustre_changelog_current_index")
+                    .with_description("current changelog index.")
+                    .build(),
+                changelog_user_index: meter
+                    .u64_gauge("lustre_changelog_user_index")
+                    .with_description("current, maximum changelog index per registered changelog user.")
+                    .build(),
+                changelog_user_idle_sec: meter
+                    .u64_gauge("lustre_changelog_user_idle_sec")
+                    .with_description("current changelog user idle seconds.")
+                    .build(),
+            }
         }
     }
-}
-
-static OST_STATS: Metric = Metric {
-    name: "lustre_oss_ost_stats",
-    help: "OSS ost stats",
-    r#type: MetricType::Gauge,
-};
 
-static OST_IO_STATS: Metric = Metric {
-    name: "lustre_oss_ost_io_stats",
-    help: "OSS ost_io stats",
-    r#type: MetricType::Gauge,
-};
-
-static OST_CREATE_STATS: Metric = Metric {
-    name: "lustre_oss_ost_create_stats",
-    help: "OSS ost_create stats",
-    r#type: MetricType::Gauge,
-};
-
-static CHANGELOG_CURRENT_INDEX: Metric = Metric {
-    name: "lustre_changelog_current_index",
-    help: "current changelog index.",
-    r#type: MetricType::Gauge,
-};
-
-static CHANGELOG_USER_INDEX: Metric = Metric {
-    name: "lustre_changelog_user_index",
-    help: "current, maximum changelog index per registered changelog user.",
-    r#type: MetricType::Gauge,
-};
-
-static CHANGELOG_USER_IDLE_SEC: Metric = Metric {
-    name: "lustre_changelog_user_idle_sec",
-    help: "current changelog user idle seconds.",
-    r#type: MetricType::Gauge,
-};
-
-fn build_oss_stats(x: OssStat, stats_map: &mut BTreeMap<&'static str, PrometheusMetric<'static>>) {
-    let OssStat { param, stats } = x;
-
-    for x in stats {
-        let Stat {
-            name,
-            units,
-            samples,
+    fn build_brw_stats(
+        x: &TargetStat<Vec<BrwStats>>,
+        otel_brw: &OpenTelemetryMetricsBrw,
+        set: &mut HashSet<(String, String, String, String)>,
```

Comment:
Use IndexSet maybe if it can help preserve ordering

Reply:
No need to order, I wanted to avoid duplicating stats (since these are counters it doubled the value due to https://github.com/whamcloud/lustrefs-exporter/issues/80#issuecomment-2710210862 )